plugins/BEdita/Core/src/Event/ImageThumbsHandler.php

```diff
@@ -65,6 +66,33 @@ public function afterSaveAssociated(Event $event): void
         $this->updateThumbs($image, $stream, $presets);
     }
 
+    /**
+     * Handle 'Thumbnails.update' and update thumbs using presets on streams
+     *
+     * @param \Cake\Event\Event $event Dispatched event.
+     * @return void
+     */
+    public function thumbnailsUpdate(Event $event): void
+    {
+        $stream = $event->getData('data');
+        if (!$stream instanceof Stream) {
+            return;
+        }
+        $objectId = $stream->get('object_id');
+        if (!$objectId) {
+            return;
+        }
+        $image = $this->fetchTable('Objects')
+            ->find('type', ['images'])
+            ->where(['id' => $objectId])
+            ->first();
```

Comment:
Maybe it's more safe to get all image data since we are going to save it.

```suggestion
        $image = $this->fetchTable('Images')
            ->find('type', ['images'])
            ->where(['id' => $objectId])
            ->first();
```